app/perfsymboltable.cpp

```diff
@@ -750,7 +750,10 @@ static PerfAddressCache::SymbolCache cacheSymbols(Dwfl_Module *module, quint64 e
         const auto symbol = dwfl_module_getsym_info(module, i, &sym, &symAddr, &shndx, &elfp, nullptr);
         if (symbol) {
             quint64 adjust = relocatedAdjust(elfp, shndx, elfStart);
-            cache.append({symAddr - elfStart, sym.st_value, sym.st_size, symbol, symbol, adjust});
+            quint64 start = sym.st_value;
+            start = (isArmArch && (start & 1)) ? start - 1 : start;
```

Comment:
we should introduce a function that takes the isArmArch param and an address and then offsets it by -1 or not, then use it here and everywhere else that currently does this manually